chalicelib/checks/wfr_checks.py

```diff
@@ -14,22 +14,258 @@
+                if pfs:
+                    if final_bam not in pfs and sample_gvcf not in pfs:
+                        raise Exception("conflicting processed files - please clean up existing processed files for sample %s" % sample_uuid)
+                    elif final_bam in pfs and sample_gvcf in pfs:
+                        continue  # already patched, do nothing.
+                ff_utils.patch_metadata({'processed_files': [final_bam, sample_gvcf]}, sample_uuid, key=ff_key)
+
+############################################################
+# patch_SNV_processed_files_to_sample_processing
+############################################################
+def patch_SNV_processed_files_to_sample_processing(metawfr_uuid, ff_key):
+    """
+    Patch sample_processing with SNV vep_vcf (full annotated vcf) and final_vcf,
+    patch completed_processes
+    """
+    ### Getting metadata
+    metawfr_meta, sp_meta, sp_meta_short = _metadata_for_patch(metawfr_uuid, ff_key)
+    ### Checking QCs are PASS and final_status completed
+    if not _eval_qcs(metawfr_meta): # if check fails, return
+        return
+
+    ### Getting uuid for files to patch and create patch body
+    patch_body = dict()
+    vep_vcf, final_vcf = '', ''
+    # get uuids for processed files
+    for wfr in metawfr_meta['workflow_runs']:
+        if wfr['name'] == 'workflow_vep-annot-check':
+            vep_vcf = wfr['output'][0]['file']['uuid']
+        elif wfr['name'] == 'workflow_hg19lo_hgvsg-check':
+            final_vcf = wfr['output'][0]['file']['uuid']
+
+    # update patch_body processed_files
+    if vep_vcf and final_vcf:
+        # this is not checking processed_files makes sense
+        #   that should be done at the check
+        try: # if processed files exist, append
+            processed_files = sp_meta_short['processed_files']
+            processed_files.append(vep_vcf)
+            processed_files.append(final_vcf)
+            patch_body = {'processed_files': processed_files}
+        except Exception: # create processed files
+            patch_body = {'processed_files': [vep_vcf, final_vcf]}
+
+    # update patch_body completed_process
+    try:
+        completed_processes = sp_meta_short['completed_processes']
+        meta_workflow_title = metawfr_meta['meta_workflow']['title']
+        if meta_workflow_title not in completed_processes:
+            completed_processes.append(meta_workflow_title)
+            patch_body.update({'completed_processes': completed_processes})
+    except Exception:
+        patch_body.update({'completed_processes': [metawfr_meta['meta_workflow']['title']]})
+
+    ### Patching metadata
+    if patch_body:
+        ff_utils.patch_metadata(patch_body, sp_uuid, key=ff_key)
+
+############################################################
+# patch_SV_processed_files_to_sample_processing
+############################################################
+def patch_SV_processed_files_to_sample_processing(metawfr_uuid, ff_key):
```

Comment:
this looks like it does almost the same thing as the SNV one above, except just checking different wfr name. These 2 functions could probably be combined and take the wfr names as an argument. If you don't want to make that change in this PR its something to keep in mind for later.